Upgrade to Redwood

Faraz32123 opened the pull request:
v18.0.0 redwood release

### Todos before merge
- [x] Ensure changelog entry exists for v18
- [x] Ensure CI/Tests are working as expected
- [x] Ensure redwood branch is up-to-date with nightly (not including the changes merged directly to redwood)

### Merge Option
- [x] Ensure "Merge Commit" option is used to merge the branch into master (Responsible: @DawoudSheraz ). This is needed to ensure the commit history when master is merged back into nightly

shadinaif committed:
feat: add atlas step to the build - FC-0012

Adding the pull behind a feature flag until it's fully implemented

Refs: FC-0012 OEP-58

shadinaif committed:
fix: typo in change log

Commit:
Merge remote-tracking branch 'origin/master' into nightly

regisb committed:
ci: remove now useless OPENEDX_RELEASE variable

regisb committed:
ci: don't even try to auto-add PRs to github project

Auto-adding PRs to the Github project is not working because the
github-token is not available there.

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
feat: upgrade dockerfiles to be compatible with 3.12.
2

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
docs: add a comment explaining the need of root user
 => [minimal  7/19] RUN echo "{}" > /openedx/config.yml                                                                          0.2s
 => ERROR [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1                     0.4s
------
 > importing cache manifest from docker.io/overhangio/openedx-discovery:17.0.0-nightly-cache:
------
------
 > [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1:
0.341 fatal: could not create work tree dir '/opt/pyenv': Permission denied
------
Dockerfile:36
--------------------
  34 |     ENV PYENV_ROOT /opt/pyenv
  35 |     # USER root
  36 | >>> RUN git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1
  37 |     # Install Python
  38 |     RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
--------------------
ERROR: failed to solve: process "/bin/sh -c git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1" did not complete successfully: exit code: 128

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
change ecommerce-worker dockerfile according to celery version 5.3.6

Commit:
Merge remote-tracking branch 'origin/master' into nightly

Commit:
add changelog entry

DawoudSheraz commented on setup.py:
```diff
@@ -34,11 +34,11 @@
-        "tutor>=17.0.0,<18.0.0",
-        "tutor-discovery>=17.0.0,<18.0.0",
-        "tutor-mfe>=17.0.0,<18.0.0",
+        "tutor>=18.0.0,<19.0.0",
+        "tutor-discovery>=18.0.0,<19.0.0",
+        "tutor-mfe>=18.0.0,<19.0.0",
```
should this not mention the PRs/branches here for now, as these versions won't be available until the actual release?

DawoudSheraz replied:
Correction: This won't cause problem on sandbox as tutor redwood will be installed before installing this branch.

Commit:
feat: upgrade to redwood

Commit:
fix: wrong mime type by adding mime-support
The admin's stylesheets e.g.https://{{ECOMMERCE_HOST}}/static/admin/css/base.css, were not loaded because its MIME type appears to be "text/plain", It should be "text/css".

Commit:
v18.0.0
